### PR TITLE
Algebraic opts: Reuse `iconst 0` from LHS

### PR DESCRIPTION
**Pull request by jameysharp**
We don't need to spend time going through the GVN map to dedup a newly-constructed `iconst 0` when we already matched that value on the left-hand side of these rules.

Also, mark these rules as subsuming any others since we can't do better than reducing an expression to a constant.

### CODE REVIEW

File: cranelift/codegen/src/opts/algebraic.isle [hunk 1]
```diff
@@ -38,13 +38,13 @@
...
+                      _
+                      zero @ (iconst ty (u64_from_imm64 0))))
+      (subsume zero))
 (rule (simplify (imul ty
-                        (iconst ty (u64_from_imm64 0))
-                        x))
-      (iconst ty (imm64 0)))
+                      zero @ (iconst ty (u64_from_imm64 0))
+                      _))
+      (subsume zero))
```
Comment (alexcrichton) [inline cranelift/codegen/src/opts/algebraic.isle:47]: Drive-by question not related to this actual PR, is this rule necessary given [this rule](https://github.com/bytecodealliance/wasmtime/blob/284fec127a3ea812deb3eabab09bffce3d7edd75/cranelift/codegen/src/opts/cprop.isle#L99-L101) that moves constants to the right?

Comment (jameysharp) [reply cranelift/codegen/src/opts/algebraic.isle:47]: Great question. I believe you're right that this rule is not necessary since we have that one. I'm not going to delete it right now though while we're still thinking through how best to express rules involving commutative operators. Chris, Nick, Trevor, and I had a long chat on the subject earlier today, but we don't have a complete plan yet.

**Review by elliottt — APPROVED** [review]
Looks good to me!

**Comment by jameysharp** [comment]
CI seems stuck, so it's time to smash random buttons and hope
